ui/store/ui.js

```diff
@@ -0,0 +1,174 @@
+// polyfills
+import '@formatjs/intl-relativetimeformat/polyfill';
+
+// dev only, "react-devtools" import is skipped in prod builds
+import 'react-devtools';
+
+import PortStream from 'extension-port-stream';
+import browser from 'webextension-polyfill';
+
+import Eth from 'ethjs';
+import EthQuery from 'eth-query';
+import StreamProvider from 'web3-stream-provider';
+import log from 'loglevel';
+import launchMetaMaskUi from '../../ui';
+import {
+  ENVIRONMENT_TYPE_FULLSCREEN,
+  ENVIRONMENT_TYPE_POPUP,
+} from '../../shared/constants/app';
+import { isManifestV3 } from '../../shared/modules/mv3.utils';
+import { SUPPORT_LINK } from '../../ui/helpers/constants/common';
+import { getErrorHtml } from '../../ui/helpers/utils/error-utils';
+import ExtensionPlatform from './platforms/extension';
+import { setupMultiplex } from './lib/stream-utils';
+import { getEnvironmentType } from './lib/util';
+import metaRPCClientFactory from './lib/metaRPCClientFactory';
+
+start().catch(log.error);
+
+async function start() {
+  async function displayCriticalError(container, err, metamaskState) {
+    const html = await getErrorHtml(SUPPORT_LINK, metamaskState);
+
+    container.innerHTML = html;
+
+    const button = document.getElementById('critical-error-button');
+
+    button.addEventListener('click', (_) => {
+      browser.runtime.reload();
+    });
+
+    log.error(err.stack);
+    throw err;
+  }
+
+  // create platform global
+  global.platform = new ExtensionPlatform();
+
+  // identify window type (popup, notification)
```

Comment:
double-checking: can the window type also be fullscreen here? If so, I would suggest adding it here or appending `, etc.` to the list of example types